### PR TITLE
[TW-762] Doc custom delay header for mock servers

### PR DESCRIPTION
**Pull request by mark-anthony-postman**
- Add `x-mock-response-delay` header
- Add autocomplete support for `x-mock` headers
- Fix vale errors

### CODE REVIEW

**Review by petewitcher — APPROVED** [review]
LGTM

**Review by MatthewPinsker-postman — APPROVED** [review]
Looks good!

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-with-api.md [hunk 1]
```diff
@@ -150,4 +150,12 @@ Use the headers `x-mock-match-request-body` or `x-mock-match-request-headers` to
...
 * To enable request body matching, set the value of `x-mock-match-request-body` to `true`.
 
-* To enable request header matching, include the header `x-mock-match-request-headers` and set its value to a comma-separated string of header keys that you want to match against the saved examples. Header matching isn't case-sensitive.
+* To enable request header matching, include the header `x-mock-match-request-headers` and set its value to a comma-separated string of header keys that you want to match in the saved examples. Header matching isn't case-sensitive.
+
+> You can also enable body and header matching in the [mock server configuration](/docs/designing-and-developing-your-api/mocking-data/setting-up-mock/#matching-request-body-and-headers).
+
+### Setting a custom response delay
+
+Use the header `x-mock-response-delay` to add a delay to the response from the mock server. You can specify a value from `0` to `180000` milliseconds. After receiving the request, the mock server waits the specified period of time before sending the response.
```
Comment (Dhembla) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-with-api.md:159]: value from `1` to `180000`

File: src/pages/docs/designing-and-developing-your-api/mocking-data/mock-with-api.md [hunk 2]
```diff
@@ -102,7 +98,7 @@ Next, add the following **raw** JSON code to the **Body** tab of the request, su
 }
 ```
 
-> By default, mock servers are publicly accessible. If you don't want the mock server to be public, add the line `"private": true` to the request body.
+> By default, mock servers are public and can receive requests from anyone. If you don't want the mock server to be public, add the line `"private": true` to the request body.
```
Comment (Dhembla) [inline src/pages/docs/designing-and-developing-your-api/mocking-data/mock-with-api.md:101]: from anyone and anywhere like code, browser, curl etc.

**Pull request merged**